### PR TITLE
Refactor config to hopefully be more sane

### PR DESCRIPTION
**Pull request by rtyley**
The config include-chain is now this (each file contains an include for the next):
- `application.conf`
- `/etc/gu/contributions.private.conf` (downloaded from S3)
- `[STAGE].public.conf` (from a stage-specific include in `contributions.private.conf`

You can download the private credentials like this:

```
$ aws s3 cp s3://contributions-private/DEV/contributions.private.conf /etc/gu/ --profile membership
```

Note that `contributions.private.conf` is now performing a double-duty:
1. Storing your private credentials!
2. Including the correct public conf for your stage: `[STAGE].public.conf`

The config files for DEV & PROD should be kept very similar in structure, so that examining them with a diffing tool can easily show any missing keys or mistakes, ie:
- `DEV.public.conf` & `PROD.public.conf` should diff well
- `DEV/contributions.private.conf` & `PROD/contributions.private.conf` also

### CODE REVIEW

File: cloud-formation/contributions-app.yml [hunk 1]
```diff
@@ -121,10 +121,14 @@
                       - " s3 cp s3://contributions-private/"
                       -
                         Ref: "Stage"
-                      - "/giraffe-keys.conf /etc/gu"
+                      - "/contributions.private.conf /etc/gu"
+
+
                 - "wget https://s3.amazonaws.com/aws-cloudwatch/downloads/latest/awslogs-agent-setup.py"
                 - "export CONF_DIR=/etc/contributions-frontend"
-                - "dpkg -i /tmp/contributions-frontend_1.0-SNAPSHfix OT_all.deb"
```
Comment (rtyley) [inline cloud-formation/contributions-app.yml:127]: This was accidentally added with https://github.com/guardian/contributions-frontend/commit/f722cfa :-)

**Comment by AWare** [comment]
:+1:

**Comment by AWare** [comment]
(We could maybe use now to put the cf in riffraff)

**Comment by rtyley** [comment]
Now this is merged to master, all devs now need to:

```
$ aws s3 cp s3://contributions-private/DEV/contributions.private.conf /etc/gu/ --profile membership
```

...and delete your old /etc/gu/giraffe-keys.conf! (at least once you're working on a branch that has this in it's history).

**Comment by rtyley** [comment]
new machines are all up:

image

...and stuff looks healthy:

image

**Pull request merged**